source/parsed_grid_generator.cc

```diff
@@ -232,62 +232,146 @@ ParsedGridGenerator<dim, spacedim>::serial()
 {
   Assert(grid_name != "", ExcNotInitialized());
-  if (grid_name == "rectangle")
-    {
-      GridGenerator::subdivided_hyper_rectangle (tria, un_int_vec_option_one,
-                                                 point_option_two, point_option_one,
-                                                 bool_option_one);
-    }
-  // TO BE DONE WHEN POSSIBLE
-  // else if(grid_name == "unit_hyperball")
-  // {
-  //   Point<spacedim> center;
-  //   double radius=1.;
-  //
-  //   GridGenerator::hyper_ball (tria,
-  //                               center, radius);
-  //   }
-  // else if(grid_name == "subhyperrectangle"){
-  //   if(dim == spacedim)
-  //   {
-  //
-  //   std::vector<unsigned int> refinement(spacedim);
-  //   for(unsigned int i=0; i<spacedim;++i)
-  //     refinement[i] = un_int_option_one;
-  //   GridGenerator::subdivided_hyper_rectangle (tria, refinement,
-  //                                    point_option_two, point_option_one);
-  //   }
-  //   else
-  //     Assert(true, ExcInternalError("dim != spacedim not supported"))
-  //
-  // }
-  else if (grid_name == "file")
-    {
-      GridIn<dim, spacedim> gi;
-      gi.attach_triangulation(tria);
-
-      std::ifstream in(input_grid_file_name.c_str());
-      AssertThrow(in, ExcIO());
-
-      std::string ext = extension(input_grid_file_name);
-      if (ext == "vtk")
-        gi.read_vtk(in);
-      else if (ext == "msh")
-        gi.read_msh(in);
-      else if (ext == "ucd" || ext == "inp")
-        gi.read_ucd(in);
-      else if (ext == "unv")
-        gi.read_unv(in);
-      else
-        Assert(false, ExcNotImplemented());
-    }
-  else
+  bool status = false;
+
+  status |= default_create_grid<dim, spacedim>( tria,
+                                                grid_name,
+                                                point_option_one,
+                                                point_option_two,
+                                                double_option_two,
+                                                double_option_one,
+                                                bool_option_one,
+                                                un_int_option_one,
+                                                un_int_vec_option_one,
+                                                input_grid_file_name);
+
+  if (!status)
+    status |= create_grid<dim, spacedim>( tria,
+                                          grid_name,
+                                          point_option_one,
+                                          point_option_two,
+                                          double_option_two,
+                                          double_option_one,
+                                          bool_option_one,
+                                          un_int_option_one,
+                                          un_int_vec_option_one);
+
+  if (!status)
     Assert(false, ExcInternalError("Unrecognized grid."));
```

Comment:
Change the message: The grid you have specified cannot be created with this combination of dim and spacedim